### ISSUE TITLE
Setting `is_adapted` to True when PreprocessingLayer is initialized with parameters

### ISSUE THREAD

**Issue by szleb** [keras-team/tf-keras#78]
**System information**.

TensorFlow version (you are using): 2.8.0
Are you willing to contribute it (Yes/No) : Yes

**Feature and the current behavior/state**.

When a PrerpocessingLayer is adapted the attribute `is_adapted` is set to True. Calling `adapt` a second time (possibly with different data) updates the parameters of the PreprocessingLayer to the new passed data (at least in some cases, e.g. with tf.keras.layers.Normalization, but does not work with tf.keras.layers.StringLookups). The user, however, can check if the PreprocessingLayer has been adapted already by looking at the attribute `is_adapted`.

When the PreprocessingLayer is instantiated in a way that all parameters are set, e.g. tf.keras.layers.Normalization with mean and variance, calling `adapt` will raise an error. But the attribute `is_adapted` is not set to `True`, so the user cannot check before calling `adapt` if the layer has been (manually) adapted already.

It would be very helpful, when the attribte `is_adapted` is set to `True` when the PreprocessingLayer has been provided with the necessary parameters when being initialized.

**Comment by sachinprasadhs** [comment]
Thank you for reporting the issue. 
We're now prioritizing [Keras 3](https://keras.io/keras_3/) development, a multi-backend framework that offers significant advantages by supporting `JAX`, `TensorFlow`, `PyTorch` and `OpenVINO`. 
This means we won't be adding new features to `TF-Keras (Keras 2)`. 

`TensorFlow` users can benefit directly from `Keras 3`, which is included by default starting with version `2.16.1`.
 We encourage you to migrate to `Keras 3` and experience its enhanced capabilities. If you feel this specific feature is crucial for the `Keras` framework itself, please open a new issue at: https://github.com/keras-team/keras/issues

For more details and usage guide about Keras 3, refer: https://keras.io/getting_started/about/
